### PR TITLE
Fixed stratosphere warm bias and code optimization for MERRA2

### PR DESCRIPTION
**Pull request by grantfirl**
## Description

Opening on behalf of @AnningCheng-NOAA. This replaces the accidentally closed https://github.com/NOAA-EMC/fv3atm/pull/664. 

This and the changes in ccpp/physics fix the stratosphere warm bias for the merra2_thompson test

These changes are expected to change baselines for the following tests:
atmaero_control_p8_rad_micro
merra2_thompson

### Issue(s) addressed

https://github.com/ufs-community/ufs-weather-model/issues/1786

## Testing

Testing was done by @AnningCheng-NOAA using UFS RTs on Hera/Intel only. Baselines will change for:
atmaero_control_p8_rad_micro
merra2_thompson

Logs will be attached in the ufs-weather-model PR.

## Dependencies

https://github.com/ufs-community/ccpp-physics/pull/88

### CODE REVIEW

**Comment by zach1221** [comment]
@grantfirl we may begin working UFS-WM PR #1815 today, would you be able to resolve the conflicts here ?

**Comment by grantfirl** [comment]
@zach1221 This is ready for final review/merge.

**Comment by zach1221** [comment]
Thanks, @grantfirl ! Sending reviews.

File: ccpp/data/CCPP_typedefs.F90 [hunk 1]
```diff
@@ -929,6 +929,8 @@ subroutine gfs_interstitial_setup_tracers(Interstitial, Model)
     if (Model%imp_physics == Model%imp_physics_thompson) then
       if (Model%ltaerosol) then
         Interstitial%nvdiff = 12
+     else if (Model%mraerosol) then
```
Comment (BrianCurtis-NOAA) [inline ccpp/data/CCPP_typedefs.F90:932]: In a future commit, please adjust this indent to match the if above it.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Comment by zach1221** [comment]
Alright, @BrianCurtis-NOAA . We have our approvals. Can you please merge this PR?

**Comment by grantfirl** [comment]
@BrianCurtis-NOAA Hold on a sec. I messed up .gitmodules.

**Comment by grantfirl** [comment]
@BrianCurtis-NOAA Sorry about that. All good now.

**Pull request merged**